bin/varnishd/cache/cache_obj.c

```diff
@@ -220,17 +222,33 @@ void
 ObjExtend(struct worker *wrk, struct objcore *oc, ssize_t l, int final)
 {
 	const struct obj_methods *om = obj_getmethods(oc);
+	AN(om->objextend);
 
 	CHECK_OBJ_NOTNULL(wrk, WORKER_MAGIC);
 	CHECK_OBJ_NOTNULL(oc->boc, BOC_MAGIC);
 	(void)final;
 	assert(l >= 0);
 
 	Lck_Lock(&oc->boc->mtx);
-	AN(om->objextend);
 	if (l > 0) {
+		/* Pass prefetch applies when only sending to one client */
+		if (oc->flags & OC_F_PRIVATE) {
```

Comment:
Having looked at #3240, I think this should be extended to cover (oc->flags & (OC_F_PRIVATE | OC_F_HFM | OC_F_HFP)) like is done there.

Reply:
The condition should rather be:

```c
if (l > 0) {
        if (oc->boc->pass_prefetch > 0) {
                assert(oc->flags & ...);
                ...
        }
        ...
}
```

Since we wipe `pass_prefetch` when it is irrelevant, it should be the condition to enter the cond loop. That would also simplify the `while` loop condition that is currently redundant. We also don't need a comment at this point, we know why we enter the nested branch.

Not commenting about the set of flags that might enable pass prefetch, I didn't look at that part.